### PR TITLE
Fix bug: Bad system call in uv__statx on Android

### PR DESCRIPTION
**Pull request by zhaozg**
Close https://github.com/libuv/libuv/issues/2282

### CODE REVIEW

File: src/unix/linux-syscalls.c [hunk 1]
```diff
@@ -358,7 +358,7 @@ int uv__statx(int dirfd,
               int flags,
               unsigned int mask,
               struct uv__statx* statxbuf) {
-#if defined(__NR_statx)
+#if defined(__NR_statx) && defined(__ANDROID__)==0
```
Comment (cjihrig) [inline src/unix/linux-syscalls.c]: Can you change the Android bits to: `!defined(__ANDROID__)`.

Comment (zhaozg) [reply src/unix/linux-syscalls.c]: fixup

File: src/unix/linux-syscalls.c [hunk 2]
```diff
@@ -358,7 +358,7 @@ int uv__statx(int dirfd,
               int flags,
               unsigned int mask,
               struct uv__statx* statxbuf) {
-#if defined(__NR_statx)
+#if defined(__NR_statx) && !defined(__ANDROID__)
```
Comment (bnoordhuis) [inline src/unix/linux-syscalls.c]: Can you also add a comment explaining why it is disabled on Android? LGTM apart from that.

**Comment by bnoordhuis** [comment]
CI: https://ci.nodejs.org/job/libuv-test-commit/1385/

**Comment by zhaozg** [comment]
Yes, CI not pass, `libuv-test-commit-freebsd` and `libuv-test-commit-windows`  fail, but not relative with this PR.

**Comment by cjihrig** [comment]
Landed in https://github.com/libuv/libuv/commit/a833f921b8860b9222d56c6c477e7891b95ebc41. Thanks!

**Pull request closed without merging**